### PR TITLE
fixed MSVC12 compilation error in test_buffers.cpp

### PR DESCRIPTION
**Pull request by cmcmurrough**
MSVC 12 complains about an ambiguous type on a couple of test checks, added (double) cast to fix.

### CODE REVIEW

File: test/io/test_buffers.cpp [hunk 1]
```diff
@@ -68,8 +68,8 @@ class BuffersTest : public ::testing::Test
         memcpy (d.data (), dptr, buffer.size () * sizeof (T));
         buffer.push (d);
         for (size_t j = 0; j < buffer.size (); ++j)
-          if (isnan (eptr[j]))
-            EXPECT_TRUE (isnan (buffer[j]));
+          if (isnan ((double) eptr[j]))
+            EXPECT_TRUE (isnan ((double) buffer[j]));
           else
```
Comment (SergioRAgostinho) [inline test/io/test_buffers.cpp]: I don't like this. ```eptr``` is of template type T and I assume ```buffer``` is something similar. It doesn't make sense to cast them always to double...

**Comment by taketwo** [comment]
I agree with what Sergio wrote. Could you try to replace `isnan` with `pcl_isnan` and see if it helps?

**Comment by SergioRAgostinho** [comment]
@UnaNancyOwen are you by any chance still running MSVC12? I have the impression the earliest you have now is MSVC13. Nevertheless can you give a go at this test and see if the compiler complains about them?

**Comment by UnaNancyOwen** [comment]
I have been tested with MSVC12 and MSVC14.
The same error has occurred in both environment.

```
1>------ Build started: Project: test_buffers, Configuration: Debug Win32 ------
1>  test_buffers.cpp
1>C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\math.h(425): error C2668: 'fpclassify' : ambiguous call to overloaded function
1>          C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\math.h(314): could be 'int fpclassify(long double)'
1>          C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\math.h(309): or       'int fpclassify(double)'
1>          C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\math.h(304): or       'int fpclassify(float)'
1>          while trying to match the argument list '(Type)'
1>          C:\pcl-1.8.0\test\io\test_buffers.cpp(71) : see reference to function template instantiation 'bool isnan<T>(_Ty)' being compiled
1>          with
1>          [
1>              T=Type
1>  ,            _Ty=Type
1>          ]
1>          C:\pcl-1.8.0\test\io\test_buffers.cpp(91) : see reference to function template instantiation 'void BuffersTest<Type>::checkBuffer<pcl::io::SingleBuffer<char>>(Buffer &,const T *,const T *,size_t)' being compiled
1>          with
1>          [
1>              Buffer=pcl::io::SingleBuffer<char>
1>  ,            T=Type
1>          ]
1>          C:\pcl-1.8.0\test\io\test_buffers.cpp(91) : see reference to function template instantiation 'void BuffersTest<Type>::checkBuffer<pcl::io::SingleBuffer<char>>(Buffer &,const T *,const T *,size_t)' being compiled
1>          with
1>          [
1>              Buffer=pcl::io::SingleBuffer<char>
1>  ,            T=Type
1>          ]
1>          C:\pcl-1.8.0\test\io\test_buffers.cpp(88) : while compiling class template member function 'void BuffersTest_SingleBuffer_Test<T>::TestBody(void)'
1>          with
1>          [
1>              T=Type
1>          ]
1>          C:\Build\googletest-release-1.7.0\include\gtest/internal/gtest-internal.h(581) : see reference to class template instantiation 'BuffersTest_SingleBuffer_Test<T>' being compiled
1>          with
1>          [
1>              T=Type
1>          ]
1>          C:\Build\googletest-release-1.7.0\include\gtest/internal/gtest-internal.h(573) : while compiling class template member function 'bool testing::internal::TypeParameterizedTest<BuffersTest,testing::internal::TemplateSel<BuffersTest_SingleBuffer_Test>,gtest_type_params_BuffersTest_>::Register(const char *,const char *,const char *,int)'
1>          C:\pcl-1.8.0\test\io\test_buffers.cpp(87) : see reference to function template instantiation 'bool testing::internal::TypeParameterizedTest<BuffersTest,testing::internal::TemplateSel<BuffersTest_SingleBuffer_Test>,gtest_type_params_BuffersTest_>::Register(const char *,const char *,const char *,int)' being compiled
1>          C:\pcl-1.8.0\test\io\test_buffers.cpp(87) : see reference to class template instantiation 'testing::internal::TypeParameterizedTest<BuffersTest,testing::internal::TemplateSel<BuffersTest_SingleBuffer_Test>,gtest_type_params_BuffersTest_>' being compiled
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

**Pull request closed without merging**